tests/run_tests.sh

```diff
@@ -0,0 +1,236 @@
+
+    for skip in ${SKIP_SCRIPTS} ${SKIP_FOLDERS}; do
+        PYTEST_COMMON_OPTS="${PYTEST_COMMON_OPTS} --ignore=${skip}"
+    done
+
+    if [[ x"${OMIT_FILE_LOG}" == x"True" ]]; then
+        UTIL_LOGGING_OPTIONS=""
+        TEST_LOGGING_OPTIONS=""
+    else
+        UTIL_LOGGING_OPTIONS="--junit-xml=logs/util.xml --log-file=logs/util.log"
+        TEST_LOGGING_OPTIONS="--junit-xml=logs/tr.xml --log-file=logs/test.log"
+    fi
+    UTIL_TOPOLOGY_OPTIONS="--topology util"
+    TEST_TOPOLOGY_OPTIONS="--topology ${TOPOLOGY}"
+}
+
+function run_dump_tests()
+{
+    echo "SCRIPT:                ${SCRIPT}"
+    echo "FULL_PATH:             ${FULL_PATH}"
+    echo "SCRIPT_PATH:           ${SCRIPT_PATH}"
+    echo "BASE_PATH:             ${BASE_PATH}"
+
+    echo "ANSIBLE_CONFIG:        ${ANSIBLE_CONFIG}"
+    echo "ANSIBLE_LIBRARY:       ${ANSIBLE_LIBRARY}"
+    echo "BYPASS_UTIL:           ${BYPASS_UTIL}"
+    echo "CLI_LOG_LEVEL:         ${CLI_LOG_LEVEL}"
+    echo "EXTRA_PARAMETERS:      ${EXTRA_PARAMETERS}"
+    echo "FILE_LOG_LEVEL:        ${FILE_LOG_LEVEL}"
+    echo "INVENTORY:             ${INVENTORY}"
+    echo "OMIT_FILE_LOG:         ${OMIT_FILE_LOG}"
+    echo "SKIP_SCRIPTS:          ${SKIP_SCRIPTS}"
+    echo "SKIP_FOLDERS:          ${SKIP_FOLDERS}"
+    echo "TEST_CASES:            ${TEST_CASES}"
+    echo "TEST_METHOD:           ${TEST_METHOD}"
+    echo "TESTBED_FILE:          ${TESTBED_FILE}"
+    echo "TEST_LOGGING_OPTIONS:  ${TEST_LOGGING_OPTIONS}"
+    echo "TEST_TOPOLOGY_OPTIONS: ${TEST_TOPOLOGY_OPTIONS}"
+    echo "UTIL_LOGGING_OPTIONS:  ${UTIL_LOGGING_OPTIONS}"
+    echo "UTIL_TOPOLOGY_OPTIONS: ${UTIL_TOPOLOGY_OPTIONS}"
+
+    echo "PYTEST_COMMON_OPTS:    ${PYTEST_COMMON_OPTS}"
+}
+
+function run_group_tests()
+{
+    echo "=== running tests in groups ==="
+    if [[ x"${BYPASS_UTIL}" == x"False" ]]; then
+        py.test ${PYTEST_COMMON_OPTS} ${UTIL_LOGGING_OPTIONS} ${UTIL_TOPOLOGY_OPTIONS} -k "not test_restart_syncd" ${EXTRA_PARAMETERS}
+    fi
+    py.test ${PYTEST_COMMON_OPTS} ${TEST_LOGGING_OPTIONS} ${TEST_TOPOLOGY_OPTIONS} -k "not test_restart_syncd" ${EXTRA_PARAMETERS} ${TEST_CASES}
+}
+
+function run_individual_tests()
+{
+    echo "=== running tests individually ==="
+    if [[ x"${BYPASS_UTIL}" == x"False" ]]; then
+        py.test ${PYTEST_COMMON_OPTS} ${UTIL_LOGGING_OPTIONS} ${UTIL_TOPOLOGY_OPTIONS} -k "not test_restart_syncd" ${EXTRA_PARAMETERS}
+    fi
+
+    sleep 120
+
+    SKIP_SCRIPTS="${SKIP_SCRIPTS} test_announce_routes.py test_nbr_health.py"
+
+    ignores=$(python -c "print '|'.join('''$SKIP_FOLDERS'''.split())")
+
+    all_scripts=$(find ./ -name 'test_*.py' | sed s:^./:: | grep -vE "^(${SKIP_FOLDERS})")
+    test_scripts=$(python -c "print '\n'.join(set('''$all_scripts'''.split()) - set('''$SKIP_SCRIPTS'''.split()))" | sort)
+
+    EXIT_CODE=0
+
+    for test_script in $test_scripts; do
+
+        tdir=$(dirname $test_script)
+        ts=$(basename $test_script)
+        tn=${ts%.py}
```

Comment:
Could you expand these to `test_script` and `test_name`? I could see this being confusing in the future.

Reply:
Done